src/main/python/ttconv/scc/scc_reader.py

```diff
@@ -145,24 +159,30 @@ def process_control_code(self, control_code: SccControlCode, time_code: SccTimeC
       # Start a new Pop-On caption
       self.current_caption = SccCaptionParagraph(self.safe_area_x_offset, self.safe_area_y_offset, SccCaptionStyle.PopOn)
 
+    if control_code is SccControlCode.RDC:
+      # Start a new Paint-On caption
+      self.set_current_to_previous()
+
+      self.current_caption = SccCaptionParagraph(self.safe_area_x_offset, self.safe_area_y_offset, SccCaptionStyle.PaintOn)
+      self.init_current_caption(time_code)
 
     if control_code in (SccControlCode.RU2, SccControlCode.RU3, SccControlCode.RU4):
       # Start a new Roll-Up caption
       self.set_current_to_previous()
 
       previous_last_lines: List[SccCaptionContent] = []
-      if self.previous_caption:
+      if len(self.previous_captions) > 0:
 
         if control_code is SccControlCode.RU2:
-          previous_last_lines = self.previous_caption.get_last_caption_lines(1)
+          previous_last_lines = self.previous_captions[0].get_last_caption_lines(1)
         if control_code is SccControlCode.RU3:
```

Comment:
Why not `elif` here?

Reply:
Similarly, why not use `elif` throughout the following function?

https://github.com/sandflow/ttconv/blob/0f013fb727a900c86ed191ff248984704d4d5e68/src/main/python/ttconv/scc/scc_reader.py#L155

Reply:
Fixed in 14f91c9